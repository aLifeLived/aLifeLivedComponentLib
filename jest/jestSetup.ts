/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
