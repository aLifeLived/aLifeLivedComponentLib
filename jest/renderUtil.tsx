import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
