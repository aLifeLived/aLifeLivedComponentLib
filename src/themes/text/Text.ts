import { createText, ResponsiveValue } from '@shopify/restyle';
import { ColorThemeTokens } from '../palette';
import { Theme } from '../theme';
import { ValueOf } from '../../ts/utils';

const FontFamilies = {
  sfUiTextBold: 'SFUIText-Bold',
  sfUiTextRegular: 'SFUIText-Regular',
  sfUiTextSemiBold: 'SFUIText-Semibold',
  sfUiDisplayRegular: 'SFUIDisplay-Regular',
};

type TextVariantToken =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'intro'
  | 'body'
  | 'bodySmall'
  | 'bodyXSmall';

type TextVariant = {
  fontFamily: ValueOf<typeof FontFamilies>;
  fontSize: number | ResponsiveValue<number, Theme>;
  lineHeight?: number | string;
  color: ColorThemeTokens;
};

export const textVariants: Record<TextVariantToken, TextVariant> = {
  h1: {
    fontSize: 52,
    lineHeight: 52,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h2: {
    fontSize: 32,
    lineHeight: 40,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h3: {
    fontSize: 24,
    lineHeight: 28.8,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h4: {
    fontSize: 20,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextRegular,
  },
  intro: {
    fontSize: 16,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextSemiBold,
  },
  body: {
    fontSize: 18,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodySmall: {
    fontSize: 16,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodyXSmall: {
    fontSize: 14,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
};

export const Text = createText<Theme>();
