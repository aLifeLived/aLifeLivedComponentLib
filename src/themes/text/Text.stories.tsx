import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { Text } from './Text';
import { CenterView } from '../../components/centerView/CenterView';

storiesOf('Text', module)
  .addDecorator(Story => <CenterView children={Story()} />)
  .add('Text', () => (
    <>
      <Text variant="h1">H1 Heading</Text>
      <Text variant="h2">H2 Heading</Text>
      <Text variant="h3">H3 Heading</Text>
      <Text variant="h4">H4 Heading</Text>
      <Text variant="body">Body</Text>
    </>
  ));
