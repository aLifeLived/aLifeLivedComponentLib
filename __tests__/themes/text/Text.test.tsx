import React from 'react';
import { Text } from '../../../src/themes/text/Text';
import { render } from '../../../jest/renderUtil';

describe('src >> themes >> Text', () => {
  describe('h1 Text render', () => {
    test('Render h1 test correctly', () => {
      const { toJSON } = render(<Text variant="h1">H1 variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('h2 Text render', () => {
    test('Render h2 test correctly', () => {
      const { toJSON } = render(<Text variant="h2">H2 variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('h3 Text render', () => {
    test('Render h3 test correctly', () => {
      const { toJSON } = render(<Text variant="h3">H3 variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('h4 Text render', () => {
    test('Render h4 test correctly', () => {
      const { toJSON } = render(<Text variant="h4">H3 variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Intro Text render', () => {
    test('Render intro test correctly', () => {
      const { toJSON } = render(<Text variant="intro">Body variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Body Text render', () => {
    test('Render body test correctly', () => {
      const { toJSON } = render(<Text variant="body">Body variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('BodySmall Text render', () => {
    test('Render BodySmall test correctly', () => {
      const { toJSON } = render(
        <Text variant="bodySmall">Caption variant</Text>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('BodyXSmall Text render', () => {
    test('Render BodyXSmall test correctly', () => {
      const { toJSON } = render(
        <Text variant="bodyXSmall">Caption variant</Text>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
