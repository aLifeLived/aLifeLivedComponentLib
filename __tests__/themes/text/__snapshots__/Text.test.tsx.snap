// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src >> themes >> Text Body Text render Render body test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIDisplay-Regular",
        "fontSize": 18,
        "lineHeight": 20,
      },
    ]
  }
>
  Body variant
</Text>
`;

exports[`src >> themes >> Text BodySmall Text render Render BodySmall test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIDisplay-Regular",
        "fontSize": 16,
        "lineHeight": 20,
      },
    ]
  }
>
  Caption variant
</Text>
`;

exports[`src >> themes >> Text BodyXSmall Text render Render BodyXSmall test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIDisplay-Regular",
        "fontSize": 14,
        "lineHeight": 20,
      },
    ]
  }
>
  Caption variant
</Text>
`;

exports[`src >> themes >> Text Intro Text render Render intro test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Semibold",
        "fontSize": 16,
        "lineHeight": 24,
      },
    ]
  }
>
  Body variant
</Text>
`;

exports[`src >> themes >> Text h1 Text render Render h1 test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Bold",
        "fontSize": 52,
        "lineHeight": 52,
      },
    ]
  }
>
  H1 variant
</Text>
`;

exports[`src >> themes >> Text h2 Text render Render h2 test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Bold",
        "fontSize": 32,
        "lineHeight": 40,
      },
    ]
  }
>
  H2 variant
</Text>
`;

exports[`src >> themes >> Text h3 Text render Render h3 test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Bold",
        "fontSize": 24,
        "lineHeight": 28.8,
      },
    ]
  }
>
  H3 variant
</Text>
`;

exports[`src >> themes >> Text h4 Text render Render h4 test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Regular",
        "fontSize": 20,
        "lineHeight": 24,
      },
    ]
  }
>
  H3 variant
</Text>
`;
